### PR TITLE
d11n/openapi test2

### PR DESCRIPTION
**Pull request by d11n**
supersedes pr #458 

-----
[View rendered CONTRIBUTING.md](https://github.com/octokit/routes/blob/d11n/openapi-test2/CONTRIBUTING.md)
[View rendered README.md](https://github.com/octokit/routes/blob/d11n/openapi-test2/README.md)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,28 +12,25 @@ Please follow these steps to report such a problem
...
-8. Figure out how to make your failing test pass. You can run a test for only
-   `*.json` file you edited based on the documentation URL by running e.g.
+7. Let us know if you’d like to work on resolving the failing test.
+   Otherwise you are done here, thank you, this is really great!
+8. Figure out how to make your failing test pass. You can run a test for just the operation you edited, based on its documentation URL, by running:
 
-   ```
-   TEST_URL=https://developer.github.com/v3/repos/#get ./node_modules/.bin/tap test/integration/single-endpoint-test.js
+   ```bash
+   $ TEST_URL=https://developer.github.com/v3/repos/#get ./node_modules/.bin/tap test/integration/endpoints-test.js
```
Comment (gr2m) [inline CONTRIBUTING.md]: I usually don’t add the `$` because it makes it harder to copy & paste. Is it needed for better syntax highlighting?

File: README.md [hunk 2]
```diff
@@ -19,105 +18,137 @@ Or install from package managers
...
+  "note": "\"number\" parameter renamed to \"issue_number\"",
+  "meta": {
+    "before": "number",
+    "after": "issue_number"
   }
 }
 ```
 
-Deprecated endpoints have a "deprecated" property
+Deprecated endpoints have a type of "idName"
```
Comment (gr2m) [inline README.md]: There are deprecations for parameter names, e.g. https://github.com/octokit/routes/blob/35d0f2e93089004bbb1e8daec3af51b89b9f76b7/openapi/api.github.com/operations/issues/list-labels-on-issue.json#L95-L105

Maybe we can rephrase it to something like 

> Changes such as deprecations are listed in the `x-changes` key?

And then show an example including the `"x-changes": [ ... ]`?

File: CONTRIBUTING.md (hunk 1 above)
Comment (d11n) [reply CONTRIBUTING.md]: pushed up a change that removes them...i do that in documentation so the context is instantly recognizable; for readability, especially since docs are often just scanned instead of read, but no big deal

File: README.md (hunk 2 above)
Comment (d11n) [reply README.md]: i have something similar in there with two `x-changes` examples...the first one is a parameter change...this one is second one...it's not as easy to see that in the diff tho...this is what's in the readme now:

---

... Deprecation information is located in the x-changes array of an operation.

Example for a deprecated parameter
```
{
  ...
}
```

Deprecated endpoints have a type of "idName"
```
{
  ...
}
```

---

let me know if this still feels wrong/off
